src/ILLink.CodeFix/DAMCodeFixProvider.cs

```diff
@@ -90,52 +90,49 @@ public sealed override FixAllProvider GetFixAllProvider ()
 			return WellKnownFixAllProviders.BatchFixer;
 		}
 
-		public override async Task RegisterCodeFixesAsync (CodeFixContext context)
+		public override Task RegisterCodeFixesAsync (CodeFixContext context)
 		{
 			var document = context.Document;
-			if (await document.GetSyntaxRootAsync (context.CancellationToken).ConfigureAwait (false) is not { } root)
-				return;
 			var diagnostic = context.Diagnostics[0];
-			if (diagnostic.AdditionalLocations.Count == 0)
-				return;
-			if (root.FindNode (diagnostic.AdditionalLocations[0].SourceSpan, getInnermostNodeForTie: true) is not SyntaxNode attributableNode)
-				return;
-			// currently not supporting multiple DAM argument, hence the check for commas in the string arguments
-			if (diagnostic.Properties[DynamicallyAccessedMembersAnalyzer.attributeArgument] is not string stringArgs || stringArgs.Contains (","))
-				return;
```

Comment:
This sort of check should be done in `RegisterCodeFixesAsync` to limit the number of "no-op" code fixes that will be offered to users. We should aim to show as few no-op fix options in normal cases as possible.